src/backend/src/controllers/users.controllers.ts

```diff
@@ -1,182 +1,95 @@
+      const settings = await UsersService.getUserSettings(userId);
 
-  if (!user) {
-    return res.status(400).json({ message: 'That user does not exist' });
+      res.status(200).json(settings);
+    } catch (error: unknown) {
+      next(error);
+    }
   }
 
-  // register a login
-  await prisma.session.create({
-    data: {
-      userId: user.userId,
-      deviceInfo: req.headers['user-agent']
-    }
-  });
+  static async updateUserSettings(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { defaultTheme, slackId } = req.body;
+      const userId = parseInt(req.params.userId);
 
-  return res.status(200).json(authenticatedUserTransformer(user));
-};
+      const settings = await UsersService.updateUserSettings(userId, defaultTheme, slackId);
 
-export const updateUserRole = async (req: Request, res: Response) => {
-  const errors = validationResult(req);
-  if (!errors.isEmpty()) {
-    return res.status(400).json({ errors: errors.array() });
+      res.status(200).json({ message: `Successfully updated settings for user ${userId}.` });
+    } catch (error: unknown) {
+      next(error);
+    }
   }
 
-  const targetUserId: number = parseInt(req.params.userId);
+  static async logUserIn(req: Request, res: Response, next: NextFunction) {
+    try {
+      const idToken = req.body.id_token;
+      const header = req.headers['user-agent'];
 
-  const { body } = req;
+      const user = await UsersService.logUserIn(idToken, header!);
 
-  const { role, userId } = body;
+      const token = generateAccessToken({ userId: user.userId, firstName: user.firstName, lastName: user.lastName });
+      res.cookie('token', token, { httpOnly: true, sameSite: 'none', secure: true });
+      res.status(200).json(user);
+    } catch (error: unknown) {
+      next(error);
+    }
+  }
 
-  const user = await prisma.user.findUnique({ where: { userId } });
+  // for dev login only!
+  static async logUserInDev(req: any, res: any, next: NextFunction) {
+    if (process.env.NODE_ENV === 'production') throw new AccessDeniedException('Cant dev login on production!');
 
-  let targetUser = await prisma.user.findUnique({ where: { userId: targetUserId } });
+    try {
+      const { userId } = req.body;
+      const header = req.headers['user-agent'];
 
-  if (!user) {
-    return res.status(404).json({ message: `user #${userId} not found!` });
-  }
+      const user = await UsersService.logUserInDev(userId, header);
 
-  if (!targetUser) {
-    return res.status(404).json({ message: `user #${targetUserId} not found!` });
+      res.status(200).json(user);
+    } catch (error: unknown) {
+      next(error);
+    }
   }
 
-  const userRole = rankUserRole(user.role);
-  const targetUserRole = rankUserRole(targetUser.role);
+  static async updateUserRole(req: Request, res: Response, next: NextFunction) {
+    try {
+      const targetUserId: number = parseInt(req.params.userId);
+      const { role, userId } = req.body;
```

Comment:
so for example, here instead of pulling out hte userId from the body you do

` const { role } = req.body; const user = getCurrentUser(res); `

then you pass that into the service and we know that the user has been authenticated